presto-cli/src/main/java/io/prestosql/cli/ClientOptions.java

```diff
@@ -366,4 +379,66 @@ public boolean equals(Object obj)
                     Objects.equals(this.value, other.value);
         }
     }
+
+    public static final class ClientExtraCredentials
+    {
+        private final String name;
+        private final String value;
+
+        public ClientExtraCredentials(String extraCredentials)
+        {
+            List<String> nameValue = NAME_VALUE_SPLITTER.splitToList(extraCredentials);
```

Comment:
`requireNonNull(extraCredentials,...);`

Reply:
@electrum is this guaranteed to be non-null? `ClientSessionProperty` and `ClientResourceEstimate` doesn't perform this check. I vaguely remember the constructor is called only we pass some non-null string via the options, but don't remember where the related code resides.

Reply:
Yes, the Airline framework calls the constructor to convert the string argument into an object, so it wouldn't make sense for it to pass a null.

Also, the null check isn't required anyway, since we are immediately using the value and thus the constructor would throw NPE. The normal reason to check for nulls in a constructor is fail immediately (maintain invariants) rather than at some point in the future when you won't know where the null came from.